apps/ui/pages/reservation-unit/[id].tsx

```diff
@@ -371,61 +417,183 @@ any): JSX.Element => {
+  searchDate,
+  searchTime,
 }: PropsNarrowed): JSX.Element | null => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
   const now = useMemo(() => new Date(), []);
   const isMobile = useMedia(`(max-width: ${breakpoints.m})`, false);
 
-  const [focusDate, setFocusDate] = useState(new Date());
   const [calendarViewType, setCalendarViewType] = useState<WeekOptions>("week");
-  const [initialReservation, setInitialReservation] =
-    useState<PendingReservation | null>(null);
-  const [isReserving, setIsReserving] = useState(false);
   const [isDialogOpen, setIsDialogOpen] = useState(false);
   const [errorMsg, setErrorMsg] = useState<string | null>(null);
-  const [shouldUnselect, setShouldUnselect] = useState(0);
-  const [storedReservation, , removeStoredReservation] =
-    useLocalStorage<PendingReservation>("reservation");
 
   const calendarRef = useRef<HTMLDivElement>(null);
-  const openPricingTermsRef = useRef<HTMLAnchorElement>(null);
-  const hash = router.asPath.split("#")[1];
 
-  const isClientATouchDevice = isTouchDevice();
-
-  useEffect(() => {
-    const scrollToCalendar = () => {
-      if (calendarRef?.current?.parentElement?.offsetTop != null) {
-        window.scroll({
-          top: calendarRef.current.parentElement.offsetTop - 20,
-          behavior: "smooth",
-        });
-      }
-    };
+  const hash = router.asPath.split("#")[1];
 
+  const reservableTimeSpans = reservationUnit?.reservableTimeSpans;
+  const todaysTimeSpans = useMemo(
+    () =>
+      reservableTimeSpans.filter(
+        (span) => span.startDatetime && isToday(new Date(span.startDatetime))
+      ),
+    [reservableTimeSpans]
+  );
+  const searchUIDate = fromUIDate(searchDate ?? "");
+  const durationOptions = useMemo(() => {
+    const {
+      minReservationDuration,
+      maxReservationDuration,
+      reservationStartInterval,
+    } = reservationUnit || {};
     if (
-      storedReservation?.reservationUnitPk === reservationUnit?.pk &&
-      storedReservation?.begin &&
-      storedReservation?.end
+      minReservationDuration == null ||
+      maxReservationDuration == null ||
+      reservationStartInterval == null
     ) {
-      setFocusDate(new Date(storedReservation.begin));
-      scrollToCalendar();
-      setInitialReservation(storedReservation);
-      removeStoredReservation();
-    } else if (hash === "calendar" && initialReservation) {
-      scrollToCalendar();
+      return [];
     }
+    return getDurationOptions(
+      minReservationDuration,
+      maxReservationDuration,
+      reservationStartInterval,
+      t
+    );
+  }, [reservationUnit, t]);
+  const minReservationDuration = reservationUnit.minReservationDuration
+    ? reservationUnit.minReservationDuration / 60
+    : 30;
+  const initialFieldValues = {
+    date:
+      searchDate && searchUIDate && isValidDate(searchUIDate)
+        ? searchDate
+        : toUIDate(new Date(todaysTimeSpans[0]?.startDatetime ?? "")),
```

Comment:
Date format? here it's in UI.

Reply:
It's supposed to be in UI format in the form/state.. `searchDate` is in UI format too..